### ISSUE TITLE
Documentation examples confusing for unique and others

### ISSUE THREAD

**Issue by IAMtheIAM** [survivejs/webpack-merge#87]
Can the docs be improved with more explanations? It assumes the person reading understands all the philosophy behind this plugin. The addition of typescript definitions would greatly improve the usability of it as well, and compatibility with typescript.

For example:
merge.unique - How do we use this? What is `customizeArray:`?

I want to replace a loader found in `baseConfig` with the loader in the new config. So replacing a typescript loader, including the `test` function.

For running karma test, the exclude property on the .ts loader will not exclude `.spec` files, but in the `baseConfig` loader, it excludes it. So how do I replace an entire module.rule for a specific `test` case?

```js

// base config
 {
            test   : /\.ts$/,
            exclude: [/\.(spec|e2e)\.ts$/],
            use    : [loaders here]
}

// new config - no exclude
    {
            test: /\.ts$/,
            exclude: [],
            use : [ loaders here] 
}
```

So basically the loaders are the same, only the Exclude property is not needed on the second one.'

Right now I couldn't figure it out so instead I used `exclude: [ENV !== 'test' ? /\.(spec|e2e)\.ts$/ : '//']` in the `baseConfig`

**Comment by bebraw** [comment]
I can accept a doc PR. To see what `unique` does, check out the test cases. The whole package is fairly thoroughly tested.

There are TypeScript definitions in a PR but someone needs to port those to actual TypeScript code still (easier to maintain).

Instead of trying to replace some loader definition, it's likely clearer to compose and go through the basic merge. If I designed the package now, I would drop most of the API and favor composition instead.

**Comment by bebraw** [comment]
I've ported the utility to TypeScript at #135 and also tuned the API. It would be great if you could comment there.

I would likely duplicate the ts configuration and push it behind a function to merge (accepts `exclude`) as granular merging tends to get tricky (i.e. altering a specific field somewhere).
